crates/wash-cli/src/up/mod.rs

```diff
@@ -355,11 +359,20 @@ pub async fn handle_up(cmd: UpCommand, output_kind: OutputKind) -> Result<Comman
     let ctx = ContextDir::new()?
         .load_default_context()
         .context("failed to load context")?;
-
+    let max_exec_time_str = cmd.wasmcloud_opts.max_execution_time.clone();
+    let max_exec_time = match max_exec_time_str {
+        Some(ref time_str) => match parse_duration(time_str) {
+            Ok(duration) => Some(duration),
+            Err(e) => {
+                eprintln!("Error parsing max-execution-time: {}", e);
+                None
+            }
+        },
+        None => None,
+    };
```

Comment:
once the value is non-optional:
```suggestion
    let max_execution_time = parse_duration(&cmd.wasmcloud_opts.max_execution_time).context("failed to parse `max-execution-time`")?;
```

if the execution time cannot be parsed, I think we should immediately